src/stratigraphy/benchmark/score.py

```diff
@@ -80,95 +55,38 @@ def get_scores(
-        overall_recall = 0
-
-    if overall_depth_interval_accuracy is None:
-        overall_depth_interval_accuracy = 0
-
-    if return_document_level_metrics:
-        return {
-            "F1": f1(overall_precision, overall_recall),
-            "precision": overall_precision,
-            "recall": overall_recall,
-            "depth_interval_accuracy": overall_depth_interval_accuracy,
-        }, pd.DataFrame(document_level_metrics)
-    else:
-        return {
-            "F1": f1(overall_precision, overall_recall),
-            "precision": overall_precision,
-            "recall": overall_recall,
-            "depth_interval_accuracy": overall_depth_interval_accuracy,
-        }
+        layer_metrics.metrics[filename] = Metrics(
+            tp=hits, fp=len(file_prediction.layers) - hits, fn=number_of_truth_values[filename] - hits
+        )
+        if depth_interval_occurences > 0:
+            depth_interval_metrics.metrics[filename] = Metrics(
+                tp=depth_interval_hits, fp=depth_interval_occurences - depth_interval_hits, fn=0
+            )
+
+    return layer_metrics, depth_interval_metrics
 
 
 def evaluate_borehole_extraction(
     predictions: dict[str, FilePredictions], number_of_truth_values: dict
-) -> tuple[dict, pd.DataFrame]:
+) -> dict[str, DatasetMetrics]:
     """Evaluate the borehole extraction predictions.
 
     Args:
        predictions (dict): The FilePredictions objects.
        number_of_truth_values (dict): The number of layer ground truth values per file.
 
     Returns:
-        tuple[dict, pd.DataFrame]: A tuple containing the overall metrics as a dictionary and the
-        individual document metrics as a DataFrame.
+        dict[str, DatasetMetrics]: A dictionary that maps a metrics name to the corresponding DatasetMetrics object
     """
-    layer_metrics, layer_document_level_metrics = evaluate_layer_extraction(predictions, number_of_truth_values)
-    (
-        metadata_metrics,
-        document_level_metrics_metadata,
-    ) = evaluate_metadata(predictions)
-    (
-        metrics_groundwater,
-        document_level_metrics_groundwater,
-        document_level_metrics_groundwater_depth,
-    ) = evaluate_groundwater(predictions)
-    metrics = {**layer_metrics, **metadata_metrics, **metrics_groundwater}
-    document_level_metrics = pd.merge(
-        layer_document_level_metrics, document_level_metrics_metadata, on="document_name", how="outer"
-    )
-    document_level_metrics = pd.merge(
-        document_level_metrics, document_level_metrics_groundwater, on="document_name", how="outer"
-    )
-    document_level_metrics = pd.merge(
-        document_level_metrics, document_level_metrics_groundwater_depth, on="document_name", how="outer"
-    )
-    return metrics, document_level_metrics
+    all_metrics = evaluate_layer_extraction(predictions, number_of_truth_values)
+    all_metrics["coordinates"] = get_metrics(predictions, "metadata_is_correct", "coordinates")
```

Comment:
Could we refactor this to get all the metrics at once?

Reply:
I'm not so sure what you mean, could you elaborate how you would refactor this?

Reply:
It was rather a small fix. What I meant: we are getting the `all_metrics` object, and then we are computing the metrics function for every single feature. To me, it seems to be nicer with the all_metrics` would directly have the `coordinates`, etc. metrics right at creation.

Reply:
We agreed not to change this now. We will likely touch this code again during the refactoring ticket that is already in progress. It is certainly true that the current implementation is not very consistent. However, another factor to consider, is that we want to keep the scoring of the different properties somewhat independent (e.g. allow scoring coordinates without necessarily also scoring everything else).